File: MultiProcess/src/main/java/com/example/multiprocess/MainActivity.java
package com.example.multiprocess;

import android.content.Intent;
import android.os.Bundle;
import android.os.Process;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
       private TextView mTv;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mTv=findViewById(R.id.test1);
        mTv.setText("MainActivity==processid==="+Process.myPid());
    }

    public void  test1(View view){
        startActivity(new Intent(this,TestActivity.class));
    }
}

File: Demo2/src/main/java/com/example/demo2/MainActivity.kt
package com.example.demo2

import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
//      registerReceiver(CustomReceiver(), IntentFilter("android.net.conn.CONNECTIVITY_CHANGE"))

    }

    fun test1(view : View){
        if(view is Button){
            view.text=Build.SERIAL
            sendBroadcast(Intent("com.example.normal.receiver"))
        }
    }
}
